src/utils.zig

```diff
@@ -66,11 +66,11 @@ pub const Compression = struct { // MARK: Compression
 		defer main.stackAllocator.free(_data);
 		var data = _data;
 		while(data.len != 0) {
-			var len = std.mem.readInt(u32, data[0..4], .big);
+			var len = std.mem.readInt(u32, data[0..4], endian);
```

Comment:
This also seems like a case we missed during #1156 
Could you also add the binary reader/writer to this one in a follow-up PR?